src/components/UserProfilePage.js

```diff
@@ -5,27 +5,24 @@ import { Card, Container, Form, ListGroup }  from 'react-bootstrap';
 import InputGroup from 'react-bootstrap/InputGroup';
 import CLIENT from '../CLIENT';
 import NoticeBanner from "./NoticeBanner";
+import { APIs } from '../api';
 
 const UserProfilePage = (props) => {
-    const [profile, setProfile] = useState(null);
+    const profile = props.profile;
+    const setProfile = props.setProfile;
     const [amount, setAmount] = useState(0.);
     const [adsPayload, setAdsPayload] = useState(null);
     const onAmount = (e) => setAmount(e.target.value);
     const [notice, setNotice] = useState("");
-    
     useEffect(() => {
-        if (profile === null) {
-            CLIENT.get("profile", {
-                headers: {
-                    'Authorization': props?.userData?.token 
-                }
-            }).then((rsp) => {
-                setProfile(rsp.data);
-            }).catch((error) => {
-                console.log(error) 
-            })
+        async function fetchProfile() {
+          const rsp = await APIs.getProfile();
+          console.log(rsp);
+          props?.setProfile(rsp);
+          localStorage.setItem('profile',JSON.stringify(rsp));
         }
-    })
+        fetchProfile();
```

Comment:
It seems the if profile == null check is needed, otherwise the frontend keeps fetching
![image](https://user-images.githubusercontent.com/34226262/207205751-3d352231-f96f-4a71-be9d-e31ae617725f.png)

Reply:
![image](https://user-images.githubusercontent.com/99834841/207206654-a2211b46-828f-4baf-817a-f291798a4ff6.png)
should be fixed